### PR TITLE
Make compatible with Python 3

### PR DESCRIPTION
**Pull request by samuelbruce**
Spent a little time making mpdsync.py compatible with Python 3. I've done only basic testing but have run it successfully under Python versions 3.12.1 and 2.7.12.

### CODE REVIEW

File: mpdsync.py [hunk 1]
```diff
@@ -48,9 +48,15 @@ class MyFloat(float):
...
-        self.roundBy = roundBy
+    if sys.version_info[0] < 3:
+        def __init__(self, num, roundBy=3):
+            super(MyFloat, self).__init__(num)
+            self.roundBy = roundBy
+    else:
+        def __new__(cls, num, roundBy=3):
+            instance = super().__new__(cls, num)
+            instance.roundBy = roundBy
+            return instance
```
Comment (alphapapa) [inline mpdsync.py:59]: Thanks for updating the call to `super()`, but why is it necessary to use `__new__` instead of `__init__`?

**Review by alphapapa — CHANGES_REQUESTED** [review]
Thanks for working on this.

**Comment by samuelbruce** [comment]
Hi Adam,

Good question! To be honest I was following a guide that I think said this
was needed because MyFloat subclasses float which is immutable. Happy for
it to be done differently if that's not right.

Kind regards,
Sam

**Comment by alphapapa** [comment]
Hi Sam,

Hm, do you have a link to that guide?  That might be correct, but I'm not sure.  It worked fine under Python 2, and I don't know if that aspect changed under Python 3.

**Comment by samuelbruce** [comment]
Hi Adam,

Here you go:
***@***.***/sub-classing-pythons-extending-built-in-immutable-classes-in-python-a06c9ecb759d

Cheers,
Sam

**Comment by alphapapa** [comment]
Unfortunately, since you replied via email, and that URL has an at-sign in it, making it appear like an email address, it appears that GitHub censored it.

**Comment by samuelbruce** [comment]
My mistake, I didn't know that would happen. [Here it is again.](https://medium.com/@wazedkhan111024/sub-classing-pythons-extending-built-in-immutable-classes-in-python-a06c9ecb759d)

File: mpdsync.py (hunk 1 above)
Comment (samuelbruce) [reply mpdsync.py:59]: I reverted the change and tested again under Python 3, and confirmed initialising the MyFloat using `__init__` causes a TypeError.

> ERROR: mpdsync: Unable to connect to master server: object.__init__() takes exactly one argument (the instance to initialize)
Traceback (most recent call last):
  File "/home/sambr/mpdsync/mpdsync.py", line 1343, in main
    master.connect()
  File "/home/sambr/mpdsync/mpdsync.py", line 277, in connect
    self.testPing()
  File "/home/sambr/mpdsync/mpdsync.py", line 477, in testPing
    self.ping()
  File "/home/sambr/mpdsync/mpdsync.py", line 225, in ping
    self.pings.insert(0, timeFunction(super(Client, self).ping))
  File "/home/sambr/mpdsync/mpdsync.py", line 124, in insert
    args = [MyFloat(a) for a in args]
            ^^^^^^^^^^
  File "/home/sambr/mpdsync/mpdsync.py", line 52, in __init__
    super(MyFloat, self).__init__(num)
TypeError: object.__init__() takes exactly one argument (the instance to initialize)

**Comment by alphapapa** [comment]
Thanks.  Looks like the somewhat-canonical explanation is here: https://www.python.org/download/releases/2.2.3/descrintro/#__new__

**Comment by alphapapa** [comment]
@samuelbruce Merged.  Thanks for working on this.

**Pull request merged**